Dockerfile

```diff
@@ -1,14 +1,77 @@
-FROM pangeo/base-image:2020.06.03
-
-# install rstudio-server
-USER root
-RUN apt-get update && \
-    curl --silent -L --fail https://download2.rstudio.org/rstudio-server-1.1.419-amd64.deb > /tmp/rstudio.deb && \
-    echo '24cd11f0405d8372b4168fc9956e0386 /tmp/rstudio.deb' | md5sum -c - && \
-    apt-get install -y /tmp/rstudio.deb && \
-    rm /tmp/rstudio.deb && \
-    apt-get clean && rm -rf /var/lib/apt/lists/*
-ENV PATH=$PATH:/usr/lib/rstudio-server/bin
-USER $NB_USER
-
-COPY CONDARC /tmp/.condarc
+# Most of the stuff here is from:
+# https://github.com/2i2c-org/pilot-hubs/tree/master/images/user
+# and
+# https://uwekorn.com/2021/03/03/deploying-conda-environments-in-docker-cheatsheet.html
+
+FROM buildpack-deps:focal-scm
+
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
+ENV NB_USER jovyan
+ENV NB_UID 1000
+
+ENV CONDA_DIR /opt/conda
+ENV R_LIBS_USER /opt/r
+
+# Explicitly add littler to PATH
+# See https://github.com/conda-forge/r-littler-feedstock/issues/6
+ENV PATH ${CONDA_DIR}/lib/R/library/littler/bin:${CONDA_DIR}/bin:$PATH
+
+RUN adduser --disabled-password --gecos "Default Jupyter user" ${NB_USER}
+
+# Create user owned R libs dir
+# This lets users temporarily install packages
+RUN mkdir -p ${R_LIBS_USER} && chown ${NB_USER}:${NB_USER} ${R_LIBS_USER}
+
+# Install these without 'recommended' packages to keep image smaller.
+# Useful utils that folks sort of take for granted
+RUN apt-get update -qq --yes > /dev/null && \
+    apt-get install --yes --no-install-recommends -qq \
+        curl \
+        git \
+        less \
+        tar \
+        tzdata \
+        vim \
+        wget \
+        locales > /dev/null
+
+RUN echo "${LC_ALL} UTF-8" > /etc/locale.gen && \
+    locale-gen
+
+WORKDIR /home/jovyan
+
+COPY install-miniforge.bash /tmp/install-miniforge.bash
+RUN /tmp/install-miniforge.bash
+
+COPY conda-linux-64.lock /tmp/conda-linux-64.lock
+RUN conda install --name base --file /tmp/conda-linux-64.lock
+
+# Needed by RStudio
+RUN apt-get update -qq --yes && \
+    apt-get install --yes --no-install-recommends -qq \
+        psmisc \
+        sudo \
+        libapparmor1 \
+        lsb-release \
+        libclang-dev  > /dev/null
+
+# Set path where R packages are installed
+# Download and install rstudio manually
+# Newer one has bug that doesn't work with jupyter-rsession-proxy
+ENV RSTUDIO_URL https://download2.rstudio.org/server/bionic/amd64/rstudio-server-1.2.5042-amd64.deb
+RUN curl --silent --location --fail ${RSTUDIO_URL} > /tmp/rstudio.deb && \
+    dpkg -i /tmp/rstudio.deb && \
+    rm /tmp/rstudio.deb
+
+USER ${NB_USER}
+
+# Install BigelowLab dev R libs
+# RUN installGithub.R BigelowLab/rasf BigelowLab/ohwobpg  # not working on GH but works locally :-/
+RUN Rscript -e "remotes::install_github('BigelowLab/ohwobpg', )" && \
+    Rscript -e "remotes::install_github('https://github.com/BigelowLab/rasf', dependencies = NA)"
```

Comment:
Those two are driving me nuts! I'm also using dependencies = NA but I need to ensure all deps are in the env.

Reply:
It looks like args get passed to `devtools::install`, so it may actually be `upgrade = FALSE`. 

> [falls back to "ask" if unset. "ask" prompts the user for which out of date packages to upgrade. For non-interactive sessions "ask" is equivalent to "always".](https://www.rdocumentation.org/packages/devtools/versions/2.4.2/topics/install)

If it's vaguely logical, it will still install dependencies, it just won't upgrade them once that's set.

Reply:
OK. Last commit is a desperate one. Added all those equals to FALSE. Also, why it is so haRd?